docs/src/main/sphinx/security/inspect-pem.rst

```diff
@@ -0,0 +1,141 @@
+.. _pem_certs:
+
+=================
+Inspect PEM files
+=================
+
+PEM certificates
+----------------
+
+PEM (Privacy Enhanced Mail) is an encoding standard used to transmit :ref:`X.509
+certificate information <glossCert>`, and :ref:`PKCS #1 public key information
+<glossKey>`. The PEM format is specified in the :ref:`PKCS #8 <glossPEM>`
+standard.
+
+Trino supports PEM-encoded certificates without a container, and also supports
+the following containers:
+
+*  :ref:`JKS <glossJKS>` keystores
+*  Binary :ref:`PKCS 12 <glossPKCS12>` containers
+
+A single PEM-encoded file can contain either certificate or key pair
+information, or both in the same file. Certified keys can contain a chain of
+certificates from successive certificate authorities. If the certificate chain
+goes all the way back to a recognized Certificate Authority, that cert does not
+need a separate trust store.
+
+Follow the steps in this topic to inspect and validate PEM-encoded key and
+certificate files. See :ref:`troubleshooting_keystore` to validate JKS keystores.
+
+.. _inspect_pems:
+
+Inspect PEM key files
+---------------------
+
+You may receive a single file that includes a private key and its certificate,
+or separate files. The file name extensions shown on this page are examples
+only; there is no extension naming standard.
+
+First, use the ``cat`` command to view these plain text files. For example:
+
+.. code-block:: shell
+
+    cat clustercoord.key
+    cat clustercoord.cert
+
+or:
+
+.. code-block:: shell
+
+    cat clustercoord.pem
+
+A key file looks something like the following:
+
+.. code-block:: text
+
+  -----BEGIN PRIVATE KEY-----
+  MIIEowIBAAKCAQEAwJL8CLeDFAHhZe3QOOF1vWt4Vuk9vyO38Y1y9SgBfB02b2jW
+  ....
+  -----END PRIVATE KEY-----
+
+If your key file reports ``BEGIN ENCRYPTED PRIVATE KEY`` instead, you must
+specify the key's password to open or inspect the file. You may have specified
+the password when requesting the key, or the password could be assigned by your
+site's network managers.
+
+  If your key file reports ``BEGIN EC PRIVATE KEY`` or
+  ``BEGIN_DSA_PRIVATE_KEY``, this designates keys using Elliptical Curve or
+  DSA alternatives to RSA. In these cases, consult ``openssl`` references for
+  the appropriate versions of the verification commands below.
+
+Inspect PEM certificate files
+-----------------------------
+
+A separate certificate file looks like the following example with the ``cat``
+command:
+
+.. code-block:: text
+
+  -----BEGIN CERTIFICATE-----
+  MIIDujCCAqICAQEwDQYJKoZIhvcNAQEFBQAwgaIxCzAJBgNVBAYTAlVTMRYwFAYD
+  ....
+  -----END CERTIFICATE-----
+  -----BEGIN CERTIFICATE-----
+  MIIDwjCCAqoCCQCxyqwZ9GK50jANBgkqhkiG9w0BAQsFADCBojELMAkGA1UEBhMC
+  ....
+  -----END CERTIFICATE-----
+
+The file can show a single certificate section, or more than one to express a
+chain of authorities, each certifying the previous.
+
+If you received a single file, make sure it shows at least one KEY and one
```

Comment:
use backticks around KEY and CERTIFICATE

Reply:
OK